mathesar/views/frontend.py

```diff
@@ -1,31 +1,105 @@
 from django.shortcuts import render, redirect
 
-from mathesar.models import Schema
-from mathesar.serializers import SchemaSerializer
-from mathesar.database.utils import get_non_default_database_keys
+from mathesar.models import Database, Schema
+from mathesar.serializers import DatabaseSerializer, SchemaSerializer
 
 
-def get_schemas(request):
+def get_schemas(request, database):
     schema_serializer = SchemaSerializer(
-        Schema.objects.all(),
+        Schema.objects.filter(database__name=database),
         many=True,
         context={'request': request}
     )
     return schema_serializer.data
 
 
-def index(request, **kwargs):
+def get_databases(request):
+    database_serializer = DatabaseSerializer(
+        Database.objects.all(),
+        many=True,
+        context={'request': request}
+    )
+    return database_serializer.data
+
+
+def get_render_info(request, **kwargs):
     requested_db = kwargs.get('dbname')
-    database_list = get_non_default_database_keys()
-    if (requested_db is None or requested_db not in database_list) and len(database_list) > 0:
-        return redirect('index', dbname=database_list[0])
+    requested_schema = kwargs.get('schema')
+    is_db_updated = False
+    is_schema_updated = False
+
+    database_list = get_databases(request)
+
+    # This block check if db is none or invalid, and if true, redirects to the first db.
+    # If db is invalid, and no dbs are configured, sets requested_db to None
+    #   - Mathesar wont start in this case, but if we plan to show a specific error page, or
+    #     allow initial start of Mathesar without a db in the future, this will be essential.
+    if (requested_db is None or next((x for x in database_list if x.get('name') == requested_db), None) is None):
```

Comment:
@pavish This code is a little hard to follow. Do you mind if I refactor it a bit?

Reply:
@kgodey Sure. It's okay with me. Thank you!